AutomatedTesting/Gem/PythonTests/editor/EditorScripts/EntityOutliner_EntityOrdering.py

```diff
@@ -0,0 +1,145 @@
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+
+class Tests:
+    entities_sorted = (
+        "Entities sorted in the expected order",
+        "Entities sorted in an incorrect order",
+    )
+
+
+def EntityOutliner_EntityOrdering():
+    """
+    Summary:
+    Verify that manual entity ordering in the entity outliner works and is stable.
+
+    Expected Behavior:
+    Several entities are created, some are manually ordered, and their order
+    is maintained, even when new entities are added.
+
+    Test Steps:
+    1) Open the empty Prefab Base level
+    2) Add 5 entities to the outliner
+    3) Move "Entity1" to the top of the order
+    4) Move "Entity4" to the bottom of the order
+    5) Add another new entity, ensure the rest of the order is unchanged
+    """
+
+    import editor_python_test_tools.pyside_utils as pyside_utils
+    import azlmbr.legacy.general as general
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    from PySide2 import QtCore, QtWidgets, QtGui, QtTest
+
+    # Grab the Editor, Entity Outliner, and Outliner Model
+    editor_window = pyside_utils.get_editor_main_window()
+    entity_outliner = pyside_utils.find_child_by_hierarchy(
+        editor_window, ..., "EntityOutlinerWidgetUI", ..., "m_objectTree"
+    )
+    entity_outliner_model = entity_outliner.model()
+
+    # Get the outliner index for the root prefab container entity
+    def get_prefab_container_index():
+        return entity_outliner_model.index(0, 0)
+
+    # Get the outlienr index for the top level entity of a given name
+    def index_for_name(name):
+        root_index = get_prefab_container_index()
+        for row in range(entity_outliner_model.rowCount(root_index)):
+            row_index = entity_outliner_model.index(row, 0, root_index)
+            if row_index.data() == name:
+                return row_index
+        return None
+
+    # Validate that the outliner top level entity order matches the expected order
+    def verify_entities_sorted(expected_order):
+        actual_order = []
+        root_index = get_prefab_container_index()
+        for row in range(entity_outliner_model.rowCount(root_index)):
+            row_index = entity_outliner_model.index(row, 0, root_index)
+            actual_order.append(row_index.data())
+
+        sorted_correctly = actual_order == expected_order
+        Report.result(Tests.entities_sorted, sorted_correctly)
+        if not sorted_correctly:
+            print(f"Expected entity order: {expected_order}")
+            print(f"Actual entity order: {actual_order}")
+
+    # Creates an entity from the outliner context menu
+    def create_entity():
+        pyside_utils.trigger_context_menu_entry(
+            entity_outliner, "Create entity", index=get_prefab_container_index()
+        )
+        general.idle_wait(0.0)
```

Comment:
just curious, what does this do?

Reply:
Added a guide comment - just forcing a 1-tick wait here to let the outliner refresh